app/client/src/components/shared/MapWidgets.tsx

```diff
@@ -1641,149 +1648,183 @@ function ShowSurroundingMonitoringLocations({
-
-        const url =
-          services.data.waterQualityPortal.monitoringLocation +
-          `search?mimeType=geojson&zip=no&bBox=${west},${south},${east},${north}`;
-        fetchCheck(url, abortController.signal)
-          .then((res: MonitoringLocationsData) => {
-            const idsToFilterOut: string[] = [];
-            const monitoringLocations = getMonitoringLocations();
-
-            if (monitoringLocations.status === 'success') {
-              monitoringLocations.data.features.forEach((location) => {
-                idsToFilterOut.push(
-                  location.properties.MonitoringLocationIdentifier,
-                );
-              });
-            }
-
-            const newData: FetchState<MonitoringLocationsData> = {
-              status: 'success',
-              data: {
-                ...res,
-                features: res.features.filter(
-                  (location) =>
-                    !idsToFilterOut.includes(
-                      location.properties.MonitoringLocationIdentifier,
-                    ),
-                ),
-              },
-            };
-            setSurroundingMonitoringLocations(newData);
-
-            const stations = buildStations(
-              newData,
-              surroundingMonitoringLocationsLayer,
-            );
-            if (!stations) return;
+        setViewReady(mapView.ready);
+        setViewStationary(mapView.stationary);
+      },
+    );
 
-            updateMonitoringLocationsLayer(
-              stations,
-              surroundingMonitoringLocationsLayer,
-            );
-          })
-          .catch((err) => {
-            if (isAbort(err)) return;
-            console.error(err);
-            setSurroundingMonitoringLocations({
-              status: 'failure',
-              data: null,
-            });
-          });
+    // keep track of layer visibility of surroundingMonitoringLocationsLayer
+    reactiveUtils.watch(
+      () => surroundingMonitoringLocationsLayer.visible,
+      () => {
+        setLayerVisible(surroundingMonitoringLocationsLayer.visible);
       },
     );
 
-    // watch for zoom changes and hide the surroundingMonitoringLocationsLayer when
-    // zoomed out too much
+    // watch for zoom changes
     reactiveUtils.watch(
       () => mapView.zoom,
       () => {
-        let newDisabledValue = false;
-        if (mapView.zoom > 8) {
-          surroundingMonitoringLocationsLayer.listMode = 'hide-children';
-        } else {
-          surroundingMonitoringLocationsLayer.listMode = 'hide';
-          newDisabledValue = true;
-        }
-
-        setDisabled(newDisabledValue);
+        setZoomLevel(mapView.zoom);
       },
     );
 
     setWatcherInitialized(true);
   }, [
+    mapView,
+    surroundingMonitoringLocationsLayer,
+    watcherInitialized,
+  ]);
+
+  const [lastExtent, setLastExtent] = useState<SimpleExtent | null>(null);
+  const [lastVisible, setLastVisible] = useState(false);
+
```

Comment:
Better to store this in a state for later access:
```suggestion
const abortController = useRef(new AbortController());
```

Reply:
Good catch! I missed that one on my refactor.